python-checks/src/test/java/org/sonar/python/checks/quickfix/PythonQuickFixVerifierTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.quickfix;
+
+import org.junit.Test;
+import org.sonar.plugins.python.api.IssueLocation;
+import org.sonar.plugins.python.api.LocationInFile;
+import org.sonar.plugins.python.api.PythonCheck;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.python.checks.ClassMethodFirstArgumentNameCheck;
+import org.sonar.python.quickfix.IssueWithQuickFix;
+import org.sonar.python.quickfix.PythonQuickFix;
+import org.sonar.python.quickfix.PythonTextEdit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class PythonQuickFixVerifierTest {
```

Comment:
I would prefer to test only the `verify` method (I don't think the other methods need to be public), create a mock implementation of `PythonCheck` and test cases such as:
* No issue is raised (we expect an `AssertionError`)
* More than 1 issue is raised
* Exactly 1 issue is raised, but it doesn't have a quickfix
* Exactly 1 issue with 1 quickfix but applying it doesn't give the expected result
* Verification is successful